bindings/py/src/CMakeLists.txt

```diff
@@ -0,0 +1,194 @@
+# -----------------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2013-2015, Numenta, Inc.  Unless you have purchased from
+# Numenta, Inc. a separate commercial license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# -----------------------------------------------------------------------------
+#
+# This builds the Python language interface.  This consists of the following libs
+# for Python2.7
+#   nupic.core.algorithms2.pyd
+#   nupic.core.engine2.pyd
+#   nupic.core.math2.pyd
+#
+# for Python3
+#   nupic.core.algorithms.pyd
+#   nupic.core.engine.pyd
+#   nupic.core.math.pyd
+#
+# For PyBind11 info See: https://pybind11.readthedocs.io/en/stable/compiling.html
+#######################################################################
+
+cmake_minimum_required(VERSION 3.11)
+project(nupic_core CXX)
+
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+set(CMAKE_VERBOSE_MAKEFILE OFF) # toggle for cmake debug
+
+
+#
+# Python support
+#
+set(src_py_algorithms_files
+    bindings/algorithms/algorithm_module.cpp
+    bindings/algorithms/py_Algorithms.cpp
+    bindings/algorithms/py_Cells4.cpp
+    bindings/algorithms/py_HTM.cpp
```

Comment:
> what is this new class? for TM?

These are all pybind11 bindings. They map Python calls into these C++ classes. They were written by @chhenning some time ago and reflect some of the logic from SWIG .i files so they may need to be updated to match changes to our C++ classes since.